File: javascript/string_reverse.js
function reverseString(str) {
    const chars = str.split('');
    let start = 0;
    let end = chars.length - 1;
    
    while (start < end) {
        const temp = chars[start];
        chars[start] = chars[end];
        chars[end] = temp;
        start++;
        end--;
    }
    
    return chars.join('');
}

function generateString(length) {
    let str = '';
    for (let i = 0; i < length; i++) {
        str += String.fromCharCode(65 + (i % 26)); // 'A' + (i % 26)
    }
    return str;
}

const args = process.argv.slice(2);
if (args.length < 1) {
    console.log('Usage: string_reverse <length>');
    process.exit(1);
}

const length = parseInt(args[0]);
if (isNaN(length)) {
    console.log('Invalid number:', args[0]);
    process.exit(1);
}

const str = generateString(length);

const start = Date.now();
const reversed = reverseString(str);
const end = Date.now();

// Calculate checksum to verify correctness
let checksum = 0;
for (let i = 0; i < reversed.length; i++) {
    checksum += reversed.charCodeAt(i);
}

const timeMs = end - start;

console.log(`JavaScript: string_reverse(${length}) = ${checksum}`);
console.log(`Time: ${timeMs}ms`);

File: python/string_reverse.py
import sys
import time

def reverse_string(s):
    chars = list(s)
    start = 0
    end = len(chars) - 1
    
    while start < end:
        chars[start], chars[end] = chars[end], chars[start]
        start += 1
        end -= 1
    
    return ''.join(chars)

def generate_string(length):
    s = ''
    for i in range(length):
        s += chr(ord('A') + (i % 26))
    return s

if len(sys.argv) < 2:
    print('Usage: string_reverse <length>')
    sys.exit(1)

try:
    length = int(sys.argv[1])
except ValueError:
    print('Invalid number:', sys.argv[1])
    sys.exit(1)

s = generate_string(length)

start = time.time()
reversed_s = reverse_string(s)
end = time.time()

# Calculate checksum to verify correctness
checksum = 0
for c in reversed_s:
    checksum += ord(c)

time_ms = int((end - start) * 1000)

print(f'Python: string_reverse({length}) = {checksum}')
print(f'Time: {time_ms}ms')